File: Data Structures & Algorithms/two-integer-sum/submission-7.go
func twoSum(nums []int, target int) []int {

	result := []int{}

	check := make(map[int]int)
	for i, v := range nums {
		diference := target - v
		if prev, ok := check[diference]; ok {
			result = append(result, prev, i)
			return result
		}
		check[v] = i
	}

	return nil
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-7.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length - 1;

        while (l < r) {

            int calc = numbers[r] + numbers[l];

            if (calc == target) {
                return new int[]{l + 1, r + 1};
            }
            if (calc > target) {
                r--;
            }
            if (calc < target) {
                l++;
            }
        }

        return new int[2];
    }
}
